HIP/hip-679.md

```diff
@@ -0,0 +1,215 @@
+The current folder structure uses an account ID per node. Account IDs are not ideal as a council member will want to change their account ID over time. To future-proof this, we will want to move from account IDs to network,shard and node ID in the path.
+
+## Rationale
+
+- There have been council members that want to change their account id but haven’t been able to. Multiple nodes can have the same node account id, in which case they will need to upload files to different bucket folders.
+- There might also be a need to recreate an account for security reasons.
+- By adapting the new node ID based path we provide the council members the flexibility to change their account IDs as well as allow only node specific access to the buckets for future community nodes inclusion.
+- Adding the network name to the path helps:
+    - Create bucket paths for ephemeral networks
+    - Combine prepod environments
+    - Allow not having to recreate buckets for resettable networks   
+- The idea of introducing rollups is to make it faster for new mirror nodes to download historical files starting from epoch. This will also help us, avoid having to rename millions of files. Renaming millions of files has to be done one by one via S3 APIs and would take months. With rollup we can create separate aggregated files while leaving individual files intact for longer period until we've verified authenticity of rollup.
+
+## User stories
+
+- As a mirror node operator, I'd like to be compatible with both the historical and new bucket path formats.
+- As a node operator, I'd like to be able to change my node's account ID.
+- As a network operator, I'd like to create the new bucket structure that supports sharding.
+- As a network operator, I'd like to merge buckets for all prepod environments and be able to use bucket paths for ephemeral networks.
+- As a network operator, I'd like to reduce cloud storage costs by eliminating duplicate files and compressing historical data.
+- As a new mirror node operator I'd like to be able to download historical data in the new aggregated format. This will improve the time it takes to startup a new mirror node with all historical data synchronized and to improve download costs to bootstrap a new mirror node.
+
+## Specification
+
+This HIP is to change the stream data bucket folder structure and in the process roll up data for faster ingestion.
+
+Current Structure:
+```
+    accountBalances/balance{nodeAccountID}/
+    eventsStreams/events_{nodeAccountID}/
+    recordstreams/record{nodeAccountID}/
+    
+Example:
+    accountBalances/balance0.0.3/
+    eventsStreams/events_0.0.3/
+    recordstreams/record0.0.3/
+```
+    
+Proposed Structure:
+```
+    {network}/{shard}/{nodeID}/balance/
+    {network}/{shard}/{nodeID}/event/
+    {network}/{shard}/{nodeID}/record/
```

Comment:
We considered adding the realm also initially. But as per Leemon, ```
we could ignore realm , as if it is ever implemented it would be invisible to streaming files.
For a round, there will be one state file per shard, and it will contain all the realms in that shard. Similarly, a record file or event file or balances file will contain info for all the realms in that shard.
So the only important factors are the ledger ID (e.g., mainnet vs public testnet), and the shard (always zero at the moment). 
```

Reply:
Ok makes sense.